src/subplots.ts

```diff
@@ -1694,19 +1720,22 @@ export class BasePlot extends PlotData {
       });
 
       canvas.addEventListener('mousedown', e => {
-        [canvasDown, frameDown, clickedObject] = this.mouseDown(canvasMouse, frameMouse);
+        [canvasDown, frameDown, clickedObject] = this.mouseDown(canvasMouse, frameMouse, absoluteMouse);
+        if (clickedObject instanceof newAxis) this.is_drawing_rubber_band = true
+        else this.is_drawing_rubber_band = false
         isDrawing = true;
       });
 
       canvas.addEventListener('mouseup', e => {
         canvas.style.cursor = 'default';
-        this.mouseUp(canvasMouse, frameMouse, canvasDown, ctrlKey);
+        console.log(e.offsetX, e.offsetY)
+        this.mouseUp(canvasMouse, frameMouse, absoluteMouse, canvasDown, ctrlKey);
         if (clickedObject) {clickedObject.mouseUp()};
-        isDrawing = false;
         this.draw();
         this.axes.forEach(axis => {axis.saveLocation()});
         this.translation = new Vertex(0, 0);
-        this.is_drawing_rubber_band = false;
+        clickedObject = undefined;
```

Comment:
Dangerous undefined assignment. Consider assigning to null and handling objects when null. undefined is a marker that something went wrong, a variable has not been passed of assigned, but developers should not set something as undefined. Weirdly enough, undefined is a different concept from "I don't want my variable to have a value"

Reply:
Done